dev/io.openliberty.jaxws.security_fat.1/fat/src/com/ibm/ws/jaxws/security/fat/ClientCertificateTest.java

```diff
@@ -0,0 +1,256 @@
+            }
+        }
+    }
+
+    @After
+    public void afterTest() throws Exception {
+        if (!dynamicUpdate) {
+            if (server != null && server.isStarted()) {
+                server.stopServer("CWPKI0023E.*", "CWWKW0601E.*", "CWPKI0022E.*", "CWWKO0801E.*"); // trust stop server to ensure server
+                                                                                                   // is stopped
+            }
+        }
+    }
+
+    // 1 Valid client certificate
+    @Mode(componenttest.custom.junit.runner.Mode.TestMode.FULL)
+    @Test
+    public void testValidClientCert() throws Exception {
+        prepareForTest("serverConfigs/" + DEFAULT_CLIENT_CERT_CONFIG, "clientCert_provider_web.xml",
+                       "bindings/validCertAlias.xml");
+
+        List<RequestParams> params = new ArrayList<>(Arrays.asList(
+                                                                   new RequestParams("employee", "pojo", SCHEMA, PORT, "/employee/employPojoService", "Hello, employee from SayHelloPojoService"),
+                                                                   new RequestParams("employee", "stateless", SCHEMA, PORT, "/employee/employStatelessService", "From other bean: Hello, employee from SayHelloStatelessService"),
+                                                                   new RequestParams("employee", "singleton", SCHEMA, PORT, "/employee/employSingletonService", "From other bean: Hello, employee from SayHelloSingletonService")));
+        runTest(params, null);
+    }
+
+    // 2 Inexistent client certificate
+    @AllowedFFDC({ "java.lang.IllegalArgumentException", "javax.net.ssl.SSLException" })
+    @Mode(componenttest.custom.junit.runner.Mode.TestMode.FULL)
+    @Test
+    public void testInexistentClientCert() throws Exception {
+        prepareForTest("serverConfigs/" + DEFAULT_CLIENT_CERT_CONFIG, "clientCert_provider_web.xml", "bindings/inexistentCertAlias.xml");
+
+        List<RequestParams> params = new ArrayList<>(Arrays.asList(
+                                                                   new RequestParams("employee", "pojo", SCHEMA, PORT, "/employee/employPojoService", "CWPKI0023E"),
+                                                                   new RequestParams("employee", "stateless", SCHEMA, PORT, "/employee/employStatelessService", "CWPKI0023E"),
+                                                                   new RequestParams("employee", "singleton", SCHEMA, PORT, "/employee/employSingletonService", "CWPKI0023E")));
+
+        runTest(params, INEXISTENT_ALIAS_SERVER_INFO);
+
+    }
+
+    // private static final String WITH_CLIENT_ALIAS_CONFIG =
+    // "withAliasClientCertConfiguration.xml";
+
+    // private static final String PATCHY_SERVER_TRUST_STORE_CONFIG =
+    // "patchyServerTrustStoreConfiguration.xml";
+
+    private static final int WAIT_TIME_OUT = 10 * 1000;
```

Comment:
Would you mind moving this field to the start of the class with the other fields?

Reply:
It was like this before my changes. But it would look tidier.